### PR TITLE
[BEAM-13015] Clarify ownership of the list for state caching across bundles

### PR DESCRIPTION
**Pull request by scwhittle**
Minor cleanup noticed while looking at code

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by scwhittle** [comment]
R: @lukecwik

**Review by lukecwik — APPROVED** [review]
Ignore this LGTM.

**Review by lukecwik — COMMENTED** [review]
Did we want to make a copy?

We could update the doc saying that this method takes ownership over the list.

Note that the two callers of this method `BagUserState` and `MultimapUserState` only have a lifespan of a single bundle so after the bundle completes they relinquish ownership over the list since `FnApiStateAccessor` clears the `stateKeyObjectCache` in finalizeState. We would definitely want to do this if we wanted to have `BagUserState` and `MultimapUserState` live beyond a single bundle.

**Review by lukecwik — CHANGES_REQUESTED** [review]
ignore earlier lgtm

**Comment by scwhittle** [comment]
R: @lukecwik updated to take ownership and added some comments, PTAL

File: sdks/java/harness/src/main/java/org/apache/beam/fn/harness/state/BagUserState.java [hunk 1]
```diff
@@ -141,6 +141,7 @@ public void asyncClose() throws Exception {
 
     // Modify the underlying cached state depending on the mutations performed
     if (isCleared) {
+      // Note this takes ownership of newValues.
```
Comment (lukecwik) [inline sdks/java/harness/src/main/java/org/apache/beam/fn/harness/state/BagUserState.java:144]:
```suggestion
      // Note this takes ownership of newValues. This object is no longer used after it has been closed.
```

**Review by lukecwik — APPROVED** [review]

**Comment by lukecwik** [comment]
Run Java PreCommit

**Comment by lukecwik** [comment]
Run Java PreCommit

File: sdks/java/harness/src/main/java/org/apache/beam/fn/harness/state/BagUserState.java [hunk 2]
```diff
@@ -141,6 +141,7 @@ public void asyncClose() throws Exception {
 
     // Modify the underlying cached state depending on the mutations performed
     if (isCleared) {
+      // Note this takes ownership of newValues. This object is no longer used after it has been closed.
```
Comment (lukecwik) [inline sdks/java/harness/src/main/java/org/apache/beam/fn/harness/state/BagUserState.java:144]:
```suggestion
      // Note this takes ownership of newValues. This object is no longer used after it has been
      // closed.
```

**Review by lukecwik — APPROVED** [review]

**Pull request merged**